php-checks/src/main/java/org/sonar/php/checks/security/SessionFixationStrategyCheck.java

```diff
@@ -0,0 +1,79 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks.security;
+
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.declaration.CallArgumentTree;
+import org.sonar.plugins.php.api.tree.expression.ArrayInitializerTree;
+import org.sonar.plugins.php.api.tree.expression.ArrayPairTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+import static org.sonar.php.checks.utils.CheckUtils.argument;
+import static org.sonar.php.checks.utils.CheckUtils.lowerCaseFunctionName;
+import static org.sonar.php.checks.utils.CheckUtils.trimQuotes;
+
+@Rule(key = "S5876")
+public class SessionFixationStrategyCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE = "Create a new session during user authentication to prevent session fixation attacks.";
+  private static final String SENSITIVE_VALUE = "none";
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String functionName = lowerCaseFunctionName(tree);
+    if (("loadfromextension".equals(functionName) && isArgumentEqualsTo(tree, "extension", 0, "security") && isArgumentSensitiveArray(tree, "values", 1)) ||
+      ("prependextensionconfig".equals(functionName) && isArgumentEqualsTo(tree, "name", 0, "security") && isArgumentSensitiveArray(tree, "config", 1))) {
+      context().newIssue(this, tree, MESSAGE);
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private static boolean isArgumentEqualsTo(FunctionCallTree functionCallTree, String name, int position, String expected) {
+    Optional<CallArgumentTree> argumentTree = argument(functionCallTree, name, position);
+    return argumentTree.isPresent() && isLiteralTreeEqualsTo(argumentTree.get().value(), expected);
+  }
+
+  private static boolean isLiteralTreeEqualsTo(ExpressionTree tree, String expected) {
+    return tree.is(Kind.REGULAR_STRING_LITERAL) && expected.equalsIgnoreCase(trimQuotes(((LiteralTree) tree).value()));
+  }
+
+  private static boolean isArgumentSensitiveArray(FunctionCallTree functionCallTree, String name, int index) {
+    return argument(functionCallTree, name, index)
+      .map(CallArgumentTree::value)
+      .map(CheckUtils::assignedValue)
+      .filter(ArrayInitializerTree.class::isInstance)
+      .map(ArrayInitializerTree.class::cast)
+      .filter(arrayTree -> arrayTree.arrayPairs()
+        .stream()
+        .anyMatch(SessionFixationStrategyCheck::checkPair))
+      .isPresent();
+  }
+
+  private static boolean checkPair(ArrayPairTree pair) {
```

Comment:
If this method returns a boolean, I would expect it to be named something like `is...`.
On the other hand, the current name would be fine if the method does not return anything and raises an issue.